init task1 the latest version

kapcta opened the pull request:
I update my project till the almost end. I don't think how I should do auto formating of coloumn size.

kapcta committed:
init task1 the latest version
I update my project till the almost end. I don't think how I should do auto formating of coloumn size.

ILevchugov commented on task1/src/main/java/ru/milovanov/task1/MultipleTable.java:
```diff
@@ -0,0 +1,36 @@
+
+public class MultipleTable {
+
+    InputWriter writer = new InputWriter();
```
Странная зависимость таблицы умножения от источника ввода данных, она вообще от него не должна зависеть

ILevchugov replied:
private final

ILevchugov commented on task1/src/main/java/ru/milovanov/task1/MultipleTable.java:
```diff
@@ -0,0 +1,36 @@
+
+    InputWriter writer = new InputWriter();
+    private final int sizeArray = writer.enterArraySize();
+    private final int[][] myArray = new int[sizeArray+1][sizeArray+1];
```
Инициализировать поля нужно в конструкторе

ILevchugov commented on task1/src/main/java/ru/milovanov/task1/MultipleTable.java:
```diff
@@ -0,0 +1,36 @@
+        return sizeArray;
+    }
+
+    public int[][] createTable() {
```
логику этого метода тоже можно вызывать в конструкторе

ILevchugov replied:
в чем вообще смысл объекта, если наружу торчит обычный массив?
примерно такое флоу: какой-то объект читает число, затем создается объект таблица умножения с этим числом и таблица умножения (не массив) передается в объект который умеет ее печатать

ILevchugov commented on task1/src/main/java/ru/milovanov/task1/OutputWriter.java:
```diff
@@ -0,0 +1,21 @@
+
+public class OutputWriter {
+
+    MultipleTable multipleTable = new MultipleTable();
```
этому место тоже в конструкторе и почему new, по идее мы в writer должны передавать какую-то уже созданную таблицу (например через конструктор) и затем вызывать метод принт

ILevchugov replied:
еще кстати private final

ILevchugov replied:
класс лучше назвать MultipleTableWriter, так понятнее что он делает

ILevchugov commented on task1/src/main/java/ru/milovanov/task1/InputWriter.java:
```diff
@@ -0,0 +1,15 @@
+import java.io.InputStreamReader;
+
+public class InputWriter {
+    BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
```
можно в конструктор, а вообще может прям в мене прочитать BufferedReader ом число )

kapcta committed:
Я что-то не понимаю суть ошибки, которую мне выдаёт эта зловещая машина... И вообще интересно как написано ок или кал
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':task1:Main.main()'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:188)
	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:263)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:186)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:174)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:109)
	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:79)
	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:79)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:74)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:408)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:395)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:388)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:374)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
Caused by: org.gradle.process.internal.ExecException: Process 'command '/Users/Mr.Milovanov/Library/Java/JavaVirtualMachines/liberica-1.8.0_333/bin/java'' finished with non-zero exit value 1

kapcta committed:
Я что-то не понимаю суть ошибки, которую мне выдаёт эта зловещая машина... И вообще интересно как написано ок или кал
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':task1:Main.main()'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:188)
	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:263)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:186)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:174)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:109)
	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:79)
	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:79)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:74)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:408)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:395)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:388)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:374)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
Caused by: org.gradle.process.internal.ExecException: Process 'command '/Users/Mr.Milovanov/Library/Java/JavaVirtualMachines/liberica-1.8.0_333/bin/java'' finished with non-zero exit value 1

ILevchugov commented on task1/src/main/java/ru/milovanov/task1/MultipleTable.java:
```diff
@@ -0,0 +1,26 @@
+public class MultipleTable {
+    private final int[][] myArray;
+    public MultipleTable(int size) throws IOException {
+        this.myArray = createTable(size);
```
npe, myArray не проинициалзированна

ILevchugov commented on task1/src/main/java/ru/milovanov/task1/MultipleTable.java:
```diff
@@ -0,0 +1,26 @@
+        this.myArray = createTable(size);
+    }
+    public int[][] createTable(int size) {
+        for (int i = 0; i <= size; i++) {
```
выйдешь за пределы массива

ILevchugov commented on task1/src/main/java/ru/milovanov/task1/OutputWriter.java:
```diff
@@ -0,0 +1,22 @@
+import java.util.HashSet;
+import java.util.Set;
+
+public class OutputWriter {
```
надо нормальный вывод написать)

ILevchugov commented on task1/src/main/java/ru/milovanov/task1/InputWriter.java:
```diff
@@ -0,0 +1,26 @@
+    public int getSize() {
+        return size;
+    }
+    public int enterArraySize() throws IOException {
```
название метода лучше readSizeFromCommandLine

ILevchugov commented on task1/src/main/java/ru/milovanov/task1/InputWriter.java:
```diff
@@ -0,0 +1,26 @@
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class InputWriter {
```
InputReader это по логике)

ILevchugov commented on task1/src/main/java/ru/milovanov/task1/MultipleTable.java:
```diff
@@ -0,0 +1,26 @@
+
+public class MultipleTable {
+    private final int[][] myArray;
+    public MultipleTable(int size) throws IOException {
```
а еще здесь не нужно throws IOException

kapcta committed:
Last version

kapcta committed:
Last version 2.0

kapcta committed:
Last version 3.0

kapcta committed:
Last version 4.0
